CONTRIBUTING.md

```diff
@@ -0,0 +1,292 @@
+8. [Commit Message Guidelines](#commit-message-conventions)
+
+### Fork the UTAM Java repo
+
+[Fork][fork-a-repo] the [salesforce/utam-java](https://github.com/salesforce/utam-java) repo. Clone your fork in your local
+workspace and [configure][configuring-a-remote-for-a-fork] your remote repository settings.
+
+```bash
+$ git clone git@github.com:<YOUR-USERNAME>/utam-java.git
+$ cd utam-java
+$ git remote add upstream git@github.com:salesforce/utam-java.git
+```
+
+### Create a feature branch
+
+```bash
+$ git checkout main
+$ git pull origin main
+$ git checkout -b <name-of-the-feature>
+```
+
+### Make your changes
+
+Modify the files, lint, format and commit your code using the following commands:
+
+```bash
+$ git add <path/to/file/to/commit>
+$ git commit
+$ git push origin <username>/<name-of-the-feature>
+```
+
+Commit your changes using a descriptive commit message that follows our [Commit Message Guidelines](#commit-message-conventions). Adherence
+to these conventions is necessary because release notes will be automatically generated from these messages. NOTE:
+optional use of _git cz_ command triggers interactive semantic commit, which prompts user with commit related questions,
+such as commit type, scope, description, and breaking changes. Use of _git cz_ is optional but recommended to ensure
+format consistency.
+
+The above commands will commit the files into your feature branch. You can keep pushing new changes into the same branch
+until you are ready to create a pull request.
+
+### Check your submission
+
+#### Test and lint your changes
+Test your changes using the unit tests by running the following command:
+
+```bash
+mvn test
+```
+
+Note that a correct submission should contain passing unit tests for the modified code.
+
+### Rebase
+
+Sometimes your feature branch will get stale with respect to the master branch, and it will require a rebase. The
+following steps can help:
+
+```bash
+$ git checkout main
+$ git pull origin main
+$ git checkout <name-of-the-feature>
+$ git rebase upstream/main
+```
+
+_note: If no conflicts arise, these commands will ensure that your changes are applied on top of the master branch. Any
+conflicts will have to be manually resolved._
+
+### Create a pull request
+
+If you've never created a pull request before, follow [these instructions][creating-a-pull-request].
+
+#### Pull Request Title
+
+A pull request title should follow [conventional commit](#commit-message-conventions) format and is automatically validated by our CI.
+
+```shell
+ex:
+commit-type(optional scope): commit description. ( NOTE: space between column and the message )
+
+Types: build, chore, ci, docs, feat, fix, perf, proposal, refactor, release, revert, style, test, wip.
+Scope: The scope should be the name of the documentation section affected (guide, tutorial, home, etc.)
+```
+
+### Update the pull request
+
+```bash
+$ git fetch origin
+$ git rebase origin/${base_branch}
```

Comment:
update usually happens against upstream i think, not origin. i might be wrong

Reply:
fixed